saq/queue/postgres.py

```diff
@@ -386,131 +349,97 @@ async def listen(
             if stop:
                 break
 
-    async def notify(self, job: Job, connection: AsyncConnection | None = None) -> None:
+    async def notify(self, job: Job, connection: PoolConnectionProxy | None = None) -> None:
         await self._notify(job.key, job.status, connection)
 
     async def update(
         self,
         job: Job,
-        connection: AsyncConnection | None = None,
+        connection: PoolConnectionProxy | None = None,
         expire_at: float | None = -1,
         **kwargs: t.Any,
     ) -> None:
         job.touched = now()
 
         for k, v in kwargs.items():
             setattr(job, k, v)
-
-        async with self.nullcontext(connection) if connection else self.pool.connection() as conn:
-            await conn.execute(
-                SQL(
-                    dedent(
-                        """
-                        UPDATE {jobs_table} SET
-                          job = %(job)s
-                          ,status = %(status)s
-                          ,scheduled = %(scheduled)s
-                          {expire_at}
-                        WHERE key = %(key)s
-                        """
-                    )
-                ).format(
-                    jobs_table=self.jobs_table,
-                    expire_at=SQL(",expire_at = %(expire_at)s" if expire_at != -1 else ""),
```

Comment:
This is cool syntax sugar I wasn't aware of from `psycopg`.  

Since there isn't a direct equivalent to this, the easiest thing to do was to break this into 2 statements.